client/my-sites/themes/marketplace-reviews-modal.tsx

```diff
@@ -0,0 +1,112 @@
+import { Dialog, Button, Card, Spinner } from '@automattic/components';
+import { TextControl, TextareaControl } from '@wordpress/components';
+import { translate } from 'i18n-calypso';
+import { useState } from 'react';
+import CardHeading from 'calypso/components/card-heading';
+import {
+	ProductType,
+	ErrorResponse,
+	useCreateMarketplaceReviewMutation,
+} from 'calypso/data/marketplace/use-marketplace-reviews';
+
+import './marketplace-reviews-modal.scss';
+
+type Props = {
+	isVisible: boolean;
+	buttons: React.ReactElement[];
+	onClose: () => void;
+	themeSlug: string;
+	themeName: string;
+};
+
+export const AddReviewsModal = ( { isVisible, buttons, onClose, themeSlug, themeName }: Props ) => {
```

Comment:
Can we make this more abstract so it can be reused for plugins? So instead of `themeSlug` it would be `slug` and instead for `themeName` it could be `name`. And moving this component to the marketplace components folder. If it's needed, we can add a `type` parameter.

Reply:
Yep, my initial approach was to have something working first and then extract it to a common area, but it is simple enough to do it now, so I will go ahead and move it 👍

Reply:
Indeed, it was a straightforward change to move it to a common area and abstract away the specific properties, here you can find the changes https://github.com/Automattic/wp-calypso/pull/84480/commits/b34ec73c8370f407352b58df5c5ca16c50597509